components/upload/upload-list.component.ts

```diff
@@ -205,7 +205,7 @@ export class NzUploadListComponent implements OnChanges {
       file.isUploading = file.status === 'uploading';
       file.message = this.genErr(file);
       file.linkProps = typeof file.linkProps === 'string' ? JSON.parse(file.linkProps) : file.linkProps;
-      file.isImageUrl = this.isImageUrl(file);
+      file.isImageUrl = file.isImageUrl ? true : this.isImageUrl(file);
```

Comment:
```suggestion
      file.isImageUrl = typeof file.isImageUrl === 'boolean' ? file.isImageUrl : this.isImageUrl(file);
```

建议使用 `typeof` 校验是否布尔类型。

Reply:
已修改